src/main/kotlin/me/serce/solidity/ide/run/compile/Solc.kt

```diff
@@ -46,42 +44,54 @@ object Solc  {
     }
   }
 
-  private fun extractSolc(classLoader: URLClassLoader): File {
-    val os = when  {
-      SystemInfoRt.isLinux -> "linux"
-      SystemInfoRt.isWindows -> "win"
-      SystemInfoRt.isMac -> "mac"
-      else -> {
-        throw IllegalStateException("Unknown OS name: ${SystemInfoRt.OS_NAME}")
+  fun extractSolc(path: String): File? {
+    URLClassLoader(SoliditySettings.getUrls(path).map { it.toUri().toURL() }.toTypedArray()).use { classLoader ->
+      val os = when {
+        SystemInfoRt.isLinux -> "linux"
+        SystemInfoRt.isWindows -> "win"
+        SystemInfoRt.isMac -> "mac"
+        else -> {
+          throw IllegalStateException("Unknown OS name: ${SystemInfoRt.OS_NAME}")
+        }
       }
-    }
-    val solcResDir = "/native/$os/solc"
-    val tmpDir = File(System.getProperty("java.io.tmpdir"), "solc")
-    tmpDir.mkdirs()
+      try {
+        val solcResDir = "/native/$os/solc"
+        val tmpDir = File(System.getProperty("java.io.tmpdir"), "solc")
+        tmpDir.mkdirs()
 
-    // for some reason, we have to load any class to be able to read any resource from the jar
-    val someClass = classLoader.loadClass("org.ethereum.util.blockchain.StandaloneBlockchain")
+        // for some reason, we have to load any class to be able to read any resource from the jar
+        val someClass = classLoader.loadClass("org.ethereum.util.blockchain.StandaloneBlockchain")
 
-    val fileList = StreamUtil.readText(someClass.getResourceAsStream("$solcResDir/file.list"), Charset.defaultCharset())
-    val files = fileList.split("\n")
-      .map { it.trim() }
-      .map {
-        val dest = File(tmpDir, it)
-        Files.copy(someClass.getResourceAsStream("$solcResDir/$it"), dest.toPath(), StandardCopyOption.REPLACE_EXISTING)
-        dest
-      }
+        val fileListStream = someClass.getResourceAsStream("$solcResDir/file.list") ?: run {
+          logger<Solc>().error("can't read file list")
+          return null
+        }
+
+        val fileList = StreamUtil.readText(fileListStream, Charset.defaultCharset())
+        val files = fileList.split("\n")
+          .map { it.trim() }
+          .map {
+            val dest = File(tmpDir, it)
+            Files.copy(someClass.getResourceAsStream("$solcResDir/$it"), dest.toPath(), StandardCopyOption.REPLACE_EXISTING)
+            dest
+          }
 
-    val solcExec = files.first()
-    solcExec.setExecutable(true)
-    return solcExec
+        val solcExec = files.first()
+        solcExec.setExecutable(true)
+        return solcExec
+      } catch (e: Exception) {
+        logger<Solc>().error("exception occurred while extracting solc", e)
```

Comment:
Even though this line is unlikely to be executed, it's better to use logger in the same way as it's used in [IntelliJ sources](https://github.com/JetBrains/intellij-community/blob/master/plugins/git4idea/src/git4idea/remote/GitConfigureRemotesDialog.kt#L68) because otherwise a new logger will be created each time.

Reply:
done